tests/python/pants_test/build_graph/test_build_file_address_mapper.py

```diff
@@ -64,6 +64,29 @@ def test_scan_addresses_with_invalid_root(self):
     with self.assertRaises(BuildFileAddressMapper.InvalidRootError):
       self.address_mapper.scan_addresses(root='subdir')
 
+  def test_determine_subproject_spec(self):
+    # Ensure that a spec referring to a subproject gets assigned to that
+    # subproject properly
```

Comment:
Grammar nit: Please end all sentences with periods (despite bad examples elsewhere in this file...).